Services/ISelectedCurrencyProvider.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orchard;
+
+namespace Nwazet.Commerce.Services {
+    public interface ISelectedCurrencyProvider : IDependency {
+        /// <summary>
+        /// The name of the provider is used to identify uniquely the provider.
+        /// </summary>
+        string Name { get; }
```

Comment:
Yes on the currency class, but this `Name` here would be the name of the provider, not of the currency